src/game/bg_misc.cpp

```diff
@@ -2918,6 +2918,29 @@ qboolean BG_WeaponIsExplosive(int weap)
 	}
 }
 
+bool BG_WeaponDisallowedInTimeruns(int weap)
```

Comment:
i think you really want to have only a single source of truth, to keep it DRY

https://github.com/etjump/etjump/blob/master/src/game/etj_utilities.cpp#L139-L153

Reply:
Yeah I should refactor that timerun section to point to this function I guess.

Reply:
On second thought, I should maybe resolve that in a seperate PR? Not really related to this, and the calls to `BG_WeaponDisallowedInTimeruns` are made during timeruns, not when starting it, so we never get double comparisons here.

Reply:
Also there are some issues with the `SelectCorrectWeapon` function anyway which I've been meaning to fix, and this function would need refactor too.

Reply:
It's merely about duplicating the list of constraints in different places, if you change it in one place the other might get unnoticed, and vice versa. Idea of supporting two identical lists is not what you really want.

Reply:
Yeah I understand, but I think it would still be better to do on a seperate commit as I would fix up that `SelectCorrectWeapon` function at the same time then, and I'd rather keep that seperate from this

Reply:
It's up to you, i didn't imply you have to refactor anything outside of this feature anyways.